### PR TITLE
Added methods and key bindings for soft clear of call list (leaves calls matching current filter)

### PR DESCRIPTION
**Pull request by dmpaul26**
I couldn't find a way to do this previously, and found some others were interested in this as well.

### CODE REVIEW

**Comment by Kaian** [comment]
HI @dmpaul26 

First of all thanks for the PR, this is a nice feature indeed!!!

Contrary to most C programs, sngrep sources use four spaces for indentation, if you don't mind changing the patch before merging it would be great. Otherwise I don't mind doing it in a later commit.

Thanks in advance!

**Comment by dmpaul26** [comment]
Ah, good to know! I fixed them in sip.c, it's the only place I noticed tabs.

File: src/curses/ui_call_list.c [hunk 1]
```diff
@@ -672,6 +673,12 @@ call_list_handle_key(ui_t *ui, int key)
                 // Clear List
                 call_list_clear(ui);
                 break;
+            case ACTION_CLEAR_CALLS_SOFT:
+            	// Remove stored calls, keeping the currently displayed calls
+            	sip_calls_clear_soft();
+            	// Clear List
+            	call_list_clear(ui);
+            	break;
```
Comment (Kaian) [inline src/curses/ui_call_list.c]: These are also tabs :)

**Comment by dmpaul26** [comment]
No, no! Strict on formatting is always good! My fault for missing that. I'm still trying to get comfortable with an IDE in linux. Did this in VIM so I didn't bother looking for a tabs = 4 spaces feature yet.

Those you pointed out are fixed now though.

**Pull request merged**